go/src/meguca/db/admin.go

```diff
@@ -372,47 +372,88 @@ func CanPerform(account, board string, action auth.ModerationLevel) (
 
 // GetSameIPPosts returns posts with the same IP and on the same board as the
 // target post
-func GetSameIPPosts(id uint64, board string) (
+func GetSameIPPosts(id uint64, board string, creds auth.SessionCreds) (
 	posts []common.StandalonePost, err error,
 ) {
-	// Get posts ids
-	r, err := sq.Select("id").
-		From("posts").
-		Where(`ip = (select ip from posts where id = ?) and board = ?`,
-			id, board).
-		Query()
-	if err != nil {
-		return
-	}
-	defer r.Close()
-	var ids = make([]uint64, 0, 64)
-	for r.Next() {
-		var id uint64
-		err = r.Scan(&id)
+	err = InTransaction(func(tx *sql.Tx) (err error) {
+		// Get posts ids
+		r, err := sq.Select("id").
+			From("posts").
+			Where(`ip = (select ip from posts where id = ?) and board = ?`,
+				id, board).
+			Query()
+
 		if err != nil {
 			return
 		}
-		ids = append(ids, id)
-	}
-	err = r.Err()
-	if err != nil {
-		return
-	}
 
-	// Read the matched posts
-	posts = make([]common.StandalonePost, 0, len(ids))
-	var post common.StandalonePost
-	for _, id := range ids {
-		post, err = GetPost(id)
-		switch err {
-		case nil:
-			posts = append(posts, post)
-		case sql.ErrNoRows: // Deleted in race
-			err = nil
-		default:
+		defer r.Close()
+		var ids = make([]uint64, 0, 64)
+
+		for r.Next() {
+			var id uint64
+			err = r.Scan(&id)
+
+			if err != nil {
+				return
+			}
+
+			ids = append(ids, id)
+		}
+
+		err = r.Err()
+
+		if err != nil {
 			return
 		}
-	}
+
+		// Read the matched posts
+		posts = make([]common.StandalonePost, 0, len(ids))
+		var post common.StandalonePost
+
+		for _, id := range ids {
+			post, err = GetPost(id)
+
+			switch err {
+			case nil:
+				posts = append(posts, post)
+			case sql.ErrNoRows: // Deleted in race
+				err = nil
+			default:
+				return
+			}
+		}
+
+		// Add a mod-log entry detailing that a meido has used meido vision
+		var name string
+		pos, err := FindPosition(board, creds.UserID)
+
+		if err != nil {
+			return
+		}
+
+		switch pos {
```

Comment:
Why did you add this reason generation logic here? Is it really needed?

Reply:
I can't say it's needed, but I thought it might be interesting to see if staff is looking at other staff.
As far as I can tell, this shouldn't leak any information and de-anonymize anything.

Reply:
An alternative would be to do "requesting ${postID} history.
I don't really think that would have any particular negative effects either.

Reply:
The other staff thing seems pointless to me. The reasons is also unneeded. You can specify the post ID already with `auth.ModLogEntry.ID`. Thoughts?

Reply:
I'll look into the post ID thing.
If you don't want the staff/reason stuff I'll remove it, if you don't care I'll keep it.
If you prefer the staff one or the postID one in the case of "don't care", please specify.

Reply:
Oh, right. Okay, so you want the postID in the modlogentry.ID spot.
I don't want to deanonymize anons, so I'll leave out the staff stuff.